conformance/client/src/main/kotlin/com/connectrpc/conformance/client/Client.kt

```diff
@@ -0,0 +1,242 @@
+        client: BidiStreamClient<Req, Resp>,
+        req: ClientCompatRequest,
+    ): ClientResponseResult {
+        TODO("implement me")
+    }
+
+    private suspend fun <Req : MessageLite, Resp : MessageLite> handleFullDuplexBidi(
+        client: BidiStreamClient<Req, Resp>,
+        req: ClientCompatRequest,
+    ): ClientResponseResult {
+        TODO("implement me")
+    }
+
+    private fun getProtocolClient(req: ClientCompatRequest): ProtocolClient {
+        // TODO: cache/re-use clients instead of creating a new one for every request
+        val serializationStrategy = serializationFactory(req.codec)
+        val scheme = if (req.serverTlsCert.isEmpty) "http" else "https"
+        val host = "$scheme://${req.host}:${req.port}"
+        var clientBuilder = OkHttpClient.Builder()
+            .protocols(listOf(asOkHttpProtocol(req.httpVersion)))
+            .connectTimeout(Duration.ofMinutes(1))
+        if (!req.serverTlsCert.isEmpty) {
+            val certs = certs(req)
+            clientBuilder = clientBuilder.sslSocketFactory(certs.sslSocketFactory(), certs.trustManager)
+        }
+        if (req.hasTimeoutMs()) {
+            clientBuilder = clientBuilder.callTimeout(Duration.ofMillis(req.timeoutMs.toLong()))
+        }
+
+        val getConfig = if (req.useGetHttpMethod) GETConfiguration.Enabled else GETConfiguration.Disabled
+        val requestCompression =
+            if (req.compression == Compression.COMPRESSION_GZIP) {
+                RequestCompression(10, GzipCompressionPool)
+            } else {
+                null
+            }
+        val compressionPools =
+            if (req.compression == Compression.COMPRESSION_GZIP) {
+                listOf(GzipCompressionPool)
+            } else {
+                emptyList()
+            }
+        return ProtocolClient(
+            httpClient = ConnectOkHttpClient(clientBuilder.build()),
+            ProtocolClientConfig(
+                host = host,
+                serializationStrategy = serializationStrategy,
+                networkProtocol = asNetworkProtocol(req.protocol),
+                getConfiguration = getConfig,
+                requestCompression = requestCompression,
+                compressionPools = compressionPools,
+            ),
+        )
+    }
+
+    private fun asNetworkProtocol(protocol: Protocol): NetworkProtocol {
+        return when (protocol) {
+            Protocol.PROTOCOL_CONNECT -> NetworkProtocol.CONNECT
+            Protocol.PROTOCOL_GRPC -> NetworkProtocol.GRPC
+            Protocol.PROTOCOL_GRPC_WEB -> NetworkProtocol.GRPC_WEB
+            else -> throw RuntimeException("unsupported protocol: $protocol")
+        }
+    }
+
+    private fun asOkHttpProtocol(httpVersion: HTTPVersion): okhttp3.Protocol {
+        return when (httpVersion) {
+            HTTPVersion.HTTP_VERSION_1 -> okhttp3.Protocol.HTTP_1_1
+            HTTPVersion.HTTP_VERSION_2 -> okhttp3.Protocol.HTTP_2
```

Comment:
If we aren't using TLS, we'll need to configure https://square.github.io/okhttp/5.x/okhttp/okhttp3/-protocol/-h2_-p-r-i-o-r_-k-n-o-w-l-e-d-g-e/index.html.